### ISSUE TITLE
View engine is being ignored in 2.9.0

### ISSUE THREAD

**Issue by jakeorr** [donpark/hbs#88]
I am using an html view engine:

```
app.set('view engine', 'html');
app.engine('html', require('./hbs')(app, config));
```

And specifying my layout (`layout_simple.html`) like this:

```
res.render('signup', {
  layout: 'layout_simple'
});
```

Up until version 2.8.0 this was working fine. After upgrading to `hbs` 2.9.0 I'm getting an error on load:

`Error: ENOENT, open '/.../layout_simple.hbs'` (where `/.../` is the correct view path.)

Specifying the layout explicitly as `layout: 'layout_simple.html'` works correctly. Seems like a bug to me.

This change may be the culprit:
https://github.com/donpark/hbs/compare/donpark:18c1823...donpark:5e44774#diff-0646323334941cee08c9e424acbee543R120

**Comment by defunctzombie** [comment]
Yep, seems to be an issue where the PR doesn't carry the extension through. Care to make a PR to fix it with a test so we don't break again?

**Comment by jakeorr** [comment]
For sure. I will take a look at this when I have a bit of time to get into it.

**Comment by defunctzombie** [comment]
fixed in 3.0.1

**Comment by jakeorr** [comment]
great news :+1:
